scripts/load_location_data.rb

```diff
@@ -0,0 +1,229 @@
+
+  combinations.shuffle.take(BUILDING_COUNT).each do |building|
+    LOCATIONS_PER_BUILDING.times do |i|
+      location_profile = profiles.sample
+      location_id = JSONModel(:location).from_hash({:building => building,
+						    :floor => (i % 2).to_s,
+						    :room => (i % 2).to_s,
+						    :area => (i % 2).to_s,
+
+						    :coordinate_1_label => 'shelf',
+						    :coordinate_1_indicator => i.to_s,
+
+                                                    :barcode => SecureRandom.hex,
+                                                    :location_profile => {
+                                                      'ref' => location_profile[:uri]
+                                                    }
+                                                   })
+                    .save
+
+      puts "Created location #{building}: #{JSONModel(:location).uri_for(location_id)}"
+
+      result << {
+	:location_uri => JSONModel(:location).uri_for(location_id),
+	:location_profile => location_profile
+      }
+    end
+  end
+
+  result
+end
+
+def create_container_profiles
+  result = {}
+
+  SAMPLE_CONTAINER_PROFILES.each do |container_profile|
+    container_profile_id = JSONModel(:container_profile).from_hash(container_profile).save
+
+    puts "Created container profile #{container_profile[:name]}: #{JSONModel(:container_profile).uri_for(container_profile_id)}"
+
+    result[container_profile[:name]] = JSONModel(:container_profile).uri_for(container_profile_id)
+  end
+
+  result
+end
+
+# Generate the appropriate container records for each location of interest
+def fill_locations_with_containers(locations, container_profiles)
+  locations.each do |location|
+    p [:location, location]
+    location_profile_name = location[:location_profile][:name]
+    packing = SAMPLE_LOCATION_PROFILE_PACKINGS.fetch(location_profile_name).sample
+
+    packing.each do |container_type, count|
+      container_profile_uri = container_profiles.fetch(container_type)
+
+      count.times do
+        top_container_id = JSONModel(:top_container).from_hash(:indicator => SecureRandom.hex,
+                                                               :barcode => SecureRandom.hex,
+                                                               :container_profile => {'ref' => container_profile_uri},
+                                                               :container_locations => [
+                                                                 {'ref' => location[:location_uri],
+                                                                  'status' => 'current',
+                                                                  'start_date' => '2000-01-01'}
+                                                               ]).save
+
+        puts "Created top container with ID: #{top_container_id}"
+      end
+    end
+  end
+end
+
+
+def main(username = 'admin', password = 'admin')
```

Comment:
Again, might be useful to make this configurable if people are not using the standard account in their dev/test setup for whatever reason.